### ISSUE TITLE
`Table` props like `striped`, `withBorder` or `withColumnBorders` don't work well with `td` `rowSpan` or `colSpan` attributes

### ISSUE THREAD

**Issue by mcmxcdev** [mantinedev/mantine#3049]
### What package has an issue

@mantine/core

### Describe the bug

The code for striping table rows taken from the DOM looks like this:

```
.mantine-1ba39vj[data-striped] tbody tr:nth-of-type(odd)
```

Which leads to wrong styling when the table uses e.g. `rowSpan` props.

***

Truncated code:

```jsx
<Table striped withBorder withColumnBorders>
  <thead>
    <tr>
      <th>Health Plan</th>
      <th>Highmark</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td rowSpan={2}>Premium</td>
      <td>Total Premium</td>
      <td>{formatAmount(plan.premium)}</td>
    </tr>
    <tr>
      <td>After allowance</td>
      <td>TODO</td>
    </tr>

    <tr>
      <td rowSpan={4}>Deductible</td>
      <td>Individual: Medical</td>
      <td>{plan.individualMedicalDeductible}</td>
    </tr>
    <tr>
      <td>Individual: Drug</td>
      <td>{plan.individualDrugDeductible}</td>
    </tr>
    <tr>
      <td>Family: Medical</td>
      <td>{plan.familyMedicalDeductible}</td>
    </tr>
    <tr>
      <td>Family: Drug</td>
      <td>{plan.familyDrugDeductible}</td>
    </tr>
  </tbody>
</Table>
```

Visualized:

![Screenshot from 2022-11-24 13-48-39](https://user-images.githubusercontent.com/16797721/203790390-c6524d3c-7c44-4566-947b-a4276e12dcea.png)

### What version of @mantine/hooks page do you have in package.json?

5.8.2

**Comment by rtivital** [comment]
Fixed in 5.9.1

**Comment by mcmxcdev** [comment]
I can confirm that the missing borders were indeed fixed in 5.9.1, but the background colors are still not alternating correctly.

The `td` with `rowSpan` is sometimes gray and other times white, with no clear pattern. You can see this on the left-most column in this picture: 
![Screenshot from 2022-12-05 08-56-36](https://user-images.githubusercontent.com/16797721/205583730-2bbcf33e-5c0c-40b1-994e-5f6ce872d481.png)

**Comment by rtivital** [comment]
Fixing background is not really possible with current implementation, as it will require using JavaScript to parse the nodes, [see example](https://stackoverflow.com/questions/42079457/how-do-you-zebra-striping-a-rowspan-table). You can remove `striped` prop and build this feature on your side.
